### ISSUE TITLE
Not pinning correctly

### ISSUE THREAD

**Issue by d3mystified** [jl94x4/ColleXions#10]
I have it set to pin 3 each from Movies and Shows but it only pins 1 for Shows.

Config: https://wtf.roflcopter.fr/paste/?31f0dd6fd5a7a38c#CrGru1FSRUf2xDZ2xcg2jjhJYrbw3HZrL1ASF5GJRAtc
Logs: https://wtf.roflcopter.fr/paste/?957f9a6f8380d0b0#EYLMS6PvJXcTgYTvSuusryJ7HrtYxLYY3ck5mgLTdid

Paste password: `JHJRxvSTM8HrtozvT`

I can't figure out why - the config does look right but this indicates that the script isn't even considering all of the configured collections:

```
2024-10-30 22:14:40,017 - INFO - Categorized collections for TV Shows: {'Shows 1': [<Collection:1294671:Best-of-1980s>, <Collection:1294672:Best-of-1990s>, <Collection:1294669:Best-of-2010s>], 'Shows 2': [], 'Shows 3': []}
```

I'm using the latest docker image.

**Comment by jl94x4** [comment]
@defluophoenix added this feature

I'll have a look into it if he doesn't get chance to before

**Comment by defluophoenix** [comment]
Yes I will have a look later today. It's definitely working for me so I'll double check the code

**Comment by jl94x4** [comment]
I made a couple of changes that may fix this, but I've created a new beta branch.

@d3mystified can you try the new branch?

```jl94x4/collexions:beta```

**Comment by jl94x4** [comment]
I've also added debugging, which will now log the count of collections available for pinning after filtering and again after enforcing the specified limit. this should tell you if its selecting enough collections to be able to pin the required amount you have chosen in your config in the logs.

**Comment by jl94x4** [comment]
Doesn't look like my changes fixed this issue @defluophoenix

**Comment by jl94x4** [comment]
Think I have made a bit of progress @d3mystified I'll let you give the beta branch a run and report back before making further changes. :)

I believe it is working as intended now.

**Comment by jl94x4** [comment]
Done lots of testing and this is definitely fixed in [v1.13](https://github.com/jl94x4/ColleXions/releases/tag/ColleXions1.13)
